webdictionary/src/main/java/com/eis0/kademlianetwork/ResourceExchangeHandler.java

```diff
@@ -1,2 +1,310 @@
 package com.eis0.kademlianetwork;
 
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.KademliaId;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handle the resource {@link Request}, it allows the user to:
+ * 1. Create a request to process a resource inside the dictionary
+ * 2. Rack up multiple pending requests, waiting to be triggered
+ * 3. Handle the response to the request, which contains the node ID closest to the resource ID that
+ * the request is trying to process
+ * 4. Process the resource, closing the pending request and removing it from the list
+ * => Resources can be added, removed, or obtained from the Dictionary
+ *
+ * @author Enrico Cestaro
+ * @author Edoardo Raimondi
+ */
+public class ResourceExchangeHandler {
+    //Maps containing the pending Requests waiting to be completed; each pending request is identified
+    //inside the Map by the request ID that the request must process
+    private Map<KademliaId, IRequest> pendingAddRequests;
+    private Map<KademliaId, IRequest> pendingGetRequests;
+    private Map<KademliaId, IRequest> pendingDeleteRequests;
+
+    public ResourceExchangeHandler() {
+        pendingAddRequests = new HashMap<>();
+        pendingGetRequests = new HashMap<>();
+        pendingDeleteRequests = new HashMap<>();
+    }
+
+    /**
+     * This method adds to the pendingAddRequests list of {@link Request} the new request, and send
+     * a message in the network asking for a receiver, that is the node with the closest ID to the
+     * resource ID
+     *
+     * @param key      The String value of the key of the resource to add to the Dictionary
+     * @param resource The String value of the resource itself to be added to the Dictionary
+     * @throws IllegalArgumentException If key or resource are null
+     */
+    public void createAddRequest(String key, String resource) {
+        if(key == null || resource == null) throw new IllegalArgumentException();
+        //Create the Request object, insert it inside the pendingAddRequests list
+        Request currentRequest = new Request(key, resource);
+        KademliaId idToFind = currentRequest.getKeyId();
+        pendingAddRequests.put(idToFind, currentRequest);
+        //Start to search for the closest ID
+        SMSPeer searcher = KademliaNetwork.getInstance().getLocalNode().getPeer();
+        processRequest(idToFind, searcher, ResearchMode.AddToDictionary);
+    }
+
+    /**
+     * This method adds to the pendingGetRequests list of {@link Request} the new request, and send
+     * a message in the network asking for a receiver, that is the node with the closest ID to the
+     * resource ID
+     *
+     * @param key The String value of the key of the resource to get from the Dictionary
+     */
+    public void createGetRequest(String key) {
```

Comment:
Missing `@throws IllegalArgumentException` specification, the same is repeated in other methods.